fabric-api-dev/src/main/java/net/fabricmc/fabric/impl/FabricDevProperties.java

```diff
@@ -0,0 +1,135 @@
+	 * Property: <code>fabric.dev.registerDebugCommands</code></br>
+	 * Default value: true</br>
+	 * {@link CommandManager#CommandManager}
+	 */
+	public static final boolean REGISTER_DEBUG_COMMANDS = getProperty("registerDebugCommands", true);
+
+	/**
+	 * Logs an error if a command threw an exception</br>
+	 * Property: <code>fabric.dev.enableCommandExceptionLogging</code></br>
+	 * Default value: true</br>
+	 * {@link CommandManager#execute(ParseResults, String)}
+	 */
+	public static final boolean ENABLE_COMMAND_EXCEPTION_LOGGING = getProperty("enableCommandExceptionLogging", true);
+
+	/**
+	 * Logs an error regarding argument ambiguity and throws an exception if an argument type is not registered</br>
+	 * Property: <code>fabric.dev.enableCommandArgumentLogging</code></br>
+	 * Default value: true</br>
+	 * {@link CommandManager#checkMissing()}
+	 */
+	public static final boolean ENABLE_COMMAND_ARGUMENT_LOGGING = getProperty("enableCommandArgumentLogging", true);
+
+	/**
+	 * Throw's an exception if a bounding box is invalid</br>
+	 * Property: <code>fabric.dev.throwOnInvalidBlockBoxes</code></br>
+	 * Default value: true</br>
+	 * {@link net.minecraft.util.math.BlockBox#BlockBox(int, int, int, int, int, int)}
+	 */
+	public static final boolean THROW_ON_INVALID_BLOCK_BOXES = getProperty("throwOnInvalidBlockBoxes", true);
+
+	/**
+	 * Logs an error if the heightmap is null</br>
+	 * Property: <code>fabric.dev.enableUnprimedHeightmapLogging</code></br>
+	 * Default value: true</br>
+	 * {@link net.minecraft.world.chunk.Chunk#sampleHeightmap(Heightmap.Type, int, int)}
+	 */
+	public static final boolean ENABLE_UNPRIMED_HEIGHTMAP_LOGGING = getProperty("enableUnprimedHeightmapLogging", true);
+
+	/**
+	 * Set's the current thread's name to the activeThreadName if debugRunnable or debugSupplier is called</br>
+	 * Property: <code>fabric.dev.enableSupplierAndRunnableDebugging</code></br>
+	 * Default value: false</br>
+	 * {@link net.minecraft.util.Util#debugRunnable(String, Runnable)} and {@link net.minecraft.util.Util#debugSupplier(String, Supplier)}
+	 */
+	public static final boolean ENABLE_SUPPLIER_AND_RUNNABLE_DEBUGGING = getProperty("enableSupplierAndRunnableDebugging", false);
+
+	/**
+	 * Invokes a method in which you should have a breakpoint to debug errors
+	 * thrown with Util#error and exceptions thrown with Util#throwOrPause</br>
+	 * Property: <code>fabric.dev.enableExceptionIdePausing</code></br>
+	 * Default value: true</br>
+	 * {@link net.minecraft.util.Util#error(String)}, {@link net.minecraft.util.Util#error(String, Throwable)}
+	 * and {@link net.minecraft.util.Util#throwOrPause(Throwable)}
+	 */
+	public static final boolean ENABLE_EXCEPTION_IDE_PAUSING = getProperty("enableExceptionIdePausing", true);
+
+	private static final boolean IS_DEVELOPMENT_ENV = FabricLoader.getInstance().isDevelopmentEnvironment();
+
+	private static boolean getProperty(String name, boolean defaultValue) {
```

Comment:
Id just disable all of these by default, we want to make a dev env as close to production as possible, at least by default.